src/analysis/openstreetmap/qgsosmdownload.h

```diff
@@ -49,6 +49,7 @@ class ANALYSIS_EXPORT QgsOSMDownload : public QObject
     static QString queryFromRect( const QgsRectangle &rect );
 
     QgsOSMDownload();
+    QgsOSMDownload( const QString &query );
```

Comment:
yes, the second is preferred.

regarding documentation, old methods without docs are expected in the test, that's why it's not failing on them, but required for any new or with a modified signature